packages/compass-collection-stats/src/stores/store.js

```diff
@@ -1,270 +1,148 @@
-     * contain all keys that you might want to modify with this.setState().
-     *
-     * @return {Object} initial store state.
-     */
-    getInitialState() {
-      return {
-        isReadonly: false,
-        isTimeSeries: false,
-        documentCount: INVALID,
-        totalDocumentSize: INVALID,
-        avgDocumentSize: INVALID,
-        indexCount: INVALID,
-        totalIndexSize: INVALID,
-        avgIndexSize: INVALID,
-        rawDocumentCount: 0,
-        rawTotalDocumentSize: 0,
-        rawAvgDocumentSize: 0,
-        rawIndexCount: 0,
-        rawTotalIndexSize: 0,
-        rawAvgIndexSize: 0
-      };
-    },
-
-    _parseCollectionDetails(result) {
-      return {
-        isReadonly: this.state.isReadonly || false,
-        isTimeSeries: this.state.isTimeSeries || false,
-        documentCount: result.document_count !== undefined ? this._format(result.document_count) : INVALID,
-        totalDocumentSize: this._format(result.document_size, 'b'),
-        avgDocumentSize: this._format(this._avg(result.document_size, result.document_count), 'b'),
-        indexCount: this._format(result.index_count),
-        totalIndexSize: this._format(result.index_size, 'b'),
-        avgIndexSize: this._format(this._avg(result.index_size, result.index_count), 'b'),
-        rawDocumentCount: result.document_count,
-        rawTotalDocumentSize: result.document_size,
-        rawAvgDocumentSize: this._avg(result.document_size, result.document_count),
-        rawIndexCount: result.index_count,
-        rawTotalIndexSize: result.index_size,
-        rawAvgIndexSize: this._avg(result.index_size, result.index_count)
-      };
-    },
-
-    _avg(size, count) {
-      if (count <= 0) {
-        return 0;
-      }
-      return size / count;
-    },
-
-    _format(value, format = 'a') {
-      const precision = value <= 1000 ? '0' : '0.0';
-      return numeral(value).format(precision + format);
-    }
-  });
-
-  // Set the app registry if preset. This must happen first.
-  if (options.localAppRegistry) {
-    const localAppRegistry = options.localAppRegistry;
-    setLocalAppRegistry(store, localAppRegistry);
-
-    /**
-     * When the collection is changed, update the store.
-     */
-    localAppRegistry.on('import-finished', () => {
-      refreshInput(store);
-    });
-
-    /**
-     * Refresh documents on data refresh.
-     */
-    localAppRegistry.on('refresh-data', () => {
-      refreshInput(store);
-    });
-
-    /**
-     * Refresh documents on document deletion.
-     */
-    localAppRegistry.on('document-deleted', () => {
-      refreshInput(store);
-    });
-
-    /**
-     * Refresh documents on document insertion.
-     */
-    localAppRegistry.on('document-inserted', () => {
-      refreshInput(store);
-    });
+const configureStore = ({ namespace, globalAppRegistry } = {}) => {
+  const { instance } =
+    globalAppRegistry.getStore('App.InstanceStore')?.getState() ?? {};
```

Comment:
Not sure if this comment is really about this PR because I see this all over compass, but does it make sense for App.InstanceStore to not exist? What really happens if it doesn't? Do we test for that? Could we write things in such a way that we don't have these kinds of tests all over? This kind of thing introduces so much complexity with extra untested code paths.

Reply:
I don't know if we can really change that kind of pattern in compass at this point. I think it is a side-effect of how everything seems to be created at once and then they all listen to events. If we had more of a react component type pattern where we only create things once all their dependencies are there (ie. it is just props/params passed to it) then we wouldn't have this sort of thing all over the place, resulting in much simpler code with fewer branches. 

Changing it bit by bit at this stage is probably not really viable as there are just countless cases of this pattern (and similar) all over. Ideally I think code flow would never have gotten anywhere near a place like this where things it depends on have any chance of not existing.